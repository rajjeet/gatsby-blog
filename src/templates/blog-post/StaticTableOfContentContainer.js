import React from 'react';
import styled from 'styled-components';
import * as theme from '../../utils/theme';

const StaticTableOfContentContainer = ({
  className, htmlContent, onClick, includeHeading,
}) => (
  <div id="static-toc" className={className}>
    {includeHeading && <h4>Outline</h4>}
    <div onClick={onClick} dangerouslySetInnerHTML={{ __html: htmlContent }} />
  </div>
);

const StyledTableOfContentContainer = styled(StaticTableOfContentContainer)`
    a {
      color: #444;
      text-decoration: none;
      :target {
        color: ${theme.primaryColor};
        font-weight: bold;
      }       
      :hover {
        color: ${theme.primaryColor};
        cursor: pointer;        
      }
    } 
    p {
      margin: 0;
    }
    ul {
      list-style: none;
    }
       
    
`;

export default StyledTableOfContentContainer;
