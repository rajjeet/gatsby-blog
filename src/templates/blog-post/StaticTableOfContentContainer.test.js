import React from 'react';
import { cleanup, render } from '@testing-library/react';
import StaticTableOfContentContainer from './StaticTableOfContentContainer';

afterEach(cleanup);

describe('StaticTableOfContentsContainer', () => {
  it('should render', () => {
    const { asFragment } = render(<StaticTableOfContentContainer htmlContent={'<h1>test</h1>'} />);
    expect(asFragment()).toMatchSnapshot();
  });

  it('should display the html markup', () => {
    const { getByText } = render(<StaticTableOfContentContainer
      htmlContent={'<h1>This blog</h1>'}
    />);
    expect(getByText('This blog')).toBeDefined();
  });

  it('should not display heading by default', () => {
    const { queryByText } = render(<StaticTableOfContentContainer
      htmlContent={'<h1>This blog</h1>'}
    />);
    expect(queryByText('Outline')).toBeNull();
  });

  it('should display heading when isHeading is set to true', () => {
    const { getByText } = render(<StaticTableOfContentContainer
      htmlContent={'<h1>This blog</h1>'}
      includeHeading
    />);
    expect(getByText('Outline')).toBeDefined();
  });
});
