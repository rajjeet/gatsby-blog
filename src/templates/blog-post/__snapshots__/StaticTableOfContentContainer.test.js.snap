// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StaticTableOfContentsContainer should render 1`] = `
<DocumentFragment>
  <div
    class="sc-bdVaJa gykhfs"
    id="static-toc"
  >
    <div>
      <h1>
        test
      </h1>
    </div>
  </div>
</DocumentFragment>
`;
